core/src/main/java/com/linecorp/armeria/server/Http2RequestDecoder.java

```diff
@@ -144,27 +146,29 @@ public void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
                 return;
             }
 
-            req = new DecodedHttpRequest(ctx.channel().eventLoop(), ++nextId, streamId,
-                                         ArmeriaHttpUtil.toArmeriaRequestHeaders(ctx, headers, endOfStream,
-                                                                                 scheme, cfg),
-                                         true, inboundTrafficController,
-                                         // FIXME(trustin): Use a different maxRequestLength
-                                         //                 for a different host.
-                                         cfg.defaultVirtualHost().maxRequestLength());
-
-            // Close the request early when it is sure that there will be
-            // neither content nor trailers.
+            final EventLoop eventLoop = ctx.channel().eventLoop();
+            final RequestHeaders armeriaRequestHeaders =
+                    ArmeriaHttpUtil.toArmeriaRequestHeaders(ctx, headers, endOfStream, scheme, cfg);
+            final int id = ++nextId;
             if (contentEmpty && endOfStream) {
```

Comment:
Question: Can we just check
`if (endOfStream) {...}`?

Reply:
That sounds nice.